src/main/java/com/google/research/bleth/simulator/IObserver.java

```diff
@@ -0,0 +1,19 @@
+package com.google.research.bleth.simulator;
+
+/** Represent an observer. */
+public interface IObserver extends Agent {
```

Comment:
Unclear naming: some of your interfaces just use an unadorned name ("Agent"), some - like this one - add an "I" prefix.
Google naming guidelines recommend the unadorned names.
In any case, please be consistent.